sdk/formrecognizer/Azure.AI.FormRecognizer/src/ClientCommon.cs

```diff
@@ -63,7 +63,12 @@ public static RequestFailedException CreateExceptionForFailedOperation(ClientDia
                 index++;
             }
 
-            return diagnostics.CreateRequestFailedException(response, errorMessage, errorCode, errorInfo);
+            return CreateRequestFailedException(diagnostics, async, response, errorMessage, errorCode, errorInfo);
         }
+
+        private static async ValueTask<RequestFailedException> CreateRequestFailedException(ClientDiagnostics diagnostics, bool async, Response response,  string errorMessage, string errorCode, Dictionary<string, string> errorInfo)
```

Comment:
This method can be later reused in other method inside this helper. But for now you can inline it to avoid complexity.

Reply:
ohhhh I see. the underlying is that I need to respect sync/async

Reply:
thank you so much for showing me this. it helps clarifying the problem I was having